### PR TITLE
enhancement / changed method to retrieve information from contentful

### PR DESCRIPTION
**Pull request by juanbravozu**
## Changes

- Changed contentful data fetch method to an Http GET request to a contentful api endpoint. This avoids looping accross all pages.

### CODE REVIEW

**Review by javiermlovo — APPROVED** [review]

File: utils/api.js [hunk 1]
```diff
@@ -6,11 +6,14 @@ const client = createClient({
 });
 
 export default async function getPageData(pageSlug) {
-  const response = await client.getEntries({ content_type: 'page' });
+  const baseUrl = `https://cdn.contentful.com/spaces/${process.env.CONTENTFUL_SPACE}/entries`;
```
Comment (Cruzlex7) [inline utils/api.js:9]: A baseURL constant is used in this case, and after you can concatenate de params.

File: utils/api.js [hunk 2]
```diff
@@ -6,11 +6,14 @@ const client = createClient({
...
+  const query = `?access_token=${process.env.CONTENTFUL_ACCESS_TOKEN}&content_type=page`;
+  const pageFilter = `&fields.slug=${pageSlug}`;
 
-  const targetPage = response.items.find((page) => page.fields.slug === pageSlug);
+  const response = await fetch(`${baseUrl}/${query}${pageFilter}`);
```
Comment (Cruzlex7) [inline utils/api.js:13]: when u use async-await instructions a try-catch block is normally used to prevent unexpected behaviors.

Comment (juanbravozu) [reply utils/api.js]: Done

File: utils/api.js (hunk 1 above)
Comment (juanbravozu) [reply utils/api.js]: Done

**Review by Cruzlex7 — APPROVED** [review]

**Pull request merged**